### PR TITLE
HPCC-12548 Roxie in memory indexes don't check for wrapping 2^32

### PR DESCRIPTION
**Pull request by richardkchapman**
Signed-off-by: Richard Chapman rchapman@hpccsystems.com

### CODE REVIEW

**Comment by richardkchapman** [comment]
@ghalliday Please review

File: roxie/ccd/ccdkey.cpp [hunk 1]
```diff
@@ -378,8 +378,10 @@ class InMemoryIndex : public CInterface, implements IInterface, implements IComp
             size32_t size = recordSize->getFixedSize();
             if (length % size)
```
Comment (ghalliday) [inline roxie/ccd/ccdkey.cpp]: @richardkchapman should this also check that (size_t)length == length?

Comment (richardkchapman) [inline roxie/ccd/ccdkey.cpp]: Not sure I see why it needs to - the limit is on there being more than 2^32 rows, not on the size being more than 2^32 bytes

Comment (ghalliday) [inline roxie/ccd/ccdkey.cpp]: you're correct, it should possibly check "length / size" is < 2^32.

Comment (richardkchapman) [inline roxie/ccd/ccdkey.cpp]: repushed to add a check

**Comment by richardkchapman** [comment]
@ghalliday Back to you

**Pull request merged**